build.gradle

```diff
@@ -1,46 +1,123 @@
+import org.grails.gradle.plugin.doc.PublishGuideTask
+
+buildscript {
+    repositories {
+        maven { url "https://repo.grails.org/grails/core" }
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.grails.grails-doc:org.grails.grails-doc.gradle.plugin:$grailsVersion"
+        classpath "org.grails:grails-gradle-plugin:$grailsVersion"
+    }
+}
+
 plugins {
-    id 'groovy'
-    id 'java-library'
-    id 'io.github.gradle-nexus.publish-plugin'
-    id 'maven-publish'
-    id 'signing'
+    id "java-library"
+    id "eclipse"
+    id "idea"
 }
 
+apply plugin: "org.grails.grails-plugin"
+apply plugin: 'org.grails.grails-doc'
+apply plugin: 'org.grails.grails-publish'
+
+
 version =  projectVersion
 group = 'org.grails.plugins'
-ext.set('grailsVersion', libs.versions.grails.get())
-ext.set('isReleaseVersion', !version.toString().endsWith('-SNAPSHOT'))
-ext.set('isSnapshot', !isReleaseVersion)
-
-apply plugin: 'org.grails.grails-plugin' // Needs to be applied after grailsVersion has been set
 
 repositories {
   mavenLocal()
   mavenCentral()
   maven { url = 'https://repo.grails.org/grails/core' }
 }
 
+java {
+    toolchain {
```

Comment:
We need to avoid toolchains because they fix the version of java and can't be matrixed in a pipeline.  The java block can be replaced with:

      compileJava.options.release = 17